### DISCUSSION TITLE
Push notification activation with 1.0 (via MAUI)

### DISCUSSION THREAD

**Discussion by chrfin** [microsoft/WindowsAppSDK#1956] [Q&A] [answered]
I just started developing an app using .NET MAUI (scheduled for release right after MAUI GA) which uses Push Notification (via Azure Notification Hub).
Last year the prototype for this app was developed using Xamarin.Forms and targeted UWP and everything worked fine there.

Now I started the "real app" and I could successfully register for notifications using `Windows.Networking.PushNotifications.PushNotificationChannelManager` identical to UWP.
I can also receive notifications via the channels `PushNotificationReceived`-event if the app is running. I also get the toast-notification if the app is not running.
BUT I could not figure out how to get the activation to work if I click the notification. Currently nothing happens if I click the notification and the app is not running and there is no `OnActivated` or `OnBackgroundActivated` method to override in the 1.0 release version (and I can't use the experimental channel with MAUI afaik).

Is there a way to get this to work with 1.0 + MAUI? Or will there be a preview of 1.1 available soon which can be used?

**Comment by nicuparente** [comment, marked as answer]
Hi Christoph:
 1. Is this "real app" also UWP or is it a Packaged Win32 app?
 2. Will you be send Toast or Raw push notifications?

RE:  Is there a way to get this to work with 1.0 + MAUI?
 - Push Notifications is only supported for C++ Win32 Packaged Applications in the 1.0-exp1 and is not part of the 1.0-GA release.

Also, I would like to note that Azure Notification Hubs currently doesn't support Windows App SDK push notifications yet.

**Reply by chrfin** [reply]
Hi @nicuparente,
thanks for your time and feedback:
1. The "real app" is based on the default .NET MAUI preview 11 template, so AFAIK it is a WinUI 3 packaged MSIX
2. I have planned to use RAW notifications and display/handle the actual toast via background tasks (as I also want to remove them easily via an additional push - in the UWP-prototype I received the RAW notification via background activation, created a toast with the Id as a tag and if I received a "delete push" removed it again via that tag)

Registration with Azure Notification Hub works fine and everything works if the app is running and also remote Toast notifications via Azure are shown if the app is not running (see https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/4449 for the problem there).
For some context: I started converting the UWP prototype to WinUI (Xamarin Forms to MAUI) before reading up the full documentation first and was not aware at that time push is not supported yet with 1.0. And everything worked fine until I hit the missing `OnActivated` and `OnBackgroundActivated` methods to override - that was the moment I started reading the docs and landed me here 😉.

So, I guess, there is no way to get this working with 1.0 GA?
Will there be a 1.1 preview available where this could work prior the 1.1 GA in Q2?

This is no blocker at the moment, as i can continue develop/test everything else, as it is working if the app is running, which is the default case during development anyhow, as it is running in the debugger...

But I would be happy to help test (or contribute some parts if possible) these features for the next version.

**Reply by AffluentOwl** [reply]
I'm a curious about how you're registering for notifications using Windows.Networking.PushNotifications with an MSIX packaged app. Is your app in the Windows Store? Or how did you bypass the part where you have to link your Package SID to the Azure Notification Hub?

Was your workflow essentially these instructions? https://docs.microsoft.com/en-us/previous-versions/azure/app-service-mobile/app-service-mobile-windows-store-dotnet-get-started-push

Also, to be clear, the Windows.Networking.PushNotifications API is different than the Microsoft.Windows.PushNotifications API which is part of the Windows App SDK, despite their very similar names.

**Reply by chrfin** [reply]
@AffluentOwl 
My workflow is based on the UWP instructions: https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-windows-store-dotnet-get-started-wns-push-notification
My app is not yet available in the store, but I have an app-registration prepared like explained in the instructions.
The store-association I solved by using the "Associate Your App with the Windows Store" wizard in an UWP app (my prototype I had anyway) and then copy over the `Package.StoreAssociation.xml` file next to the `Package.appxmanifest` in the MAUI project.

Is there already any documentation for `Microsoft.Windows.PushNotifications` aside from here: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/notifications/push/?

**Reply by AffluentOwl** [reply]
Thanks for the link. In my case, I wanted push notifications in a non-UWP app, so I'll have to wait for WindowsAppSDK's next release, which will release PushNotifications for all app types. I too found it difficult to find the official documentation since this feature is still experimental it is missing from the official [docs](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/), the best bet seems to be this [spec](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/PushNotifications/PushNotifications-spec.md) (especially the example and API outline at the bottom).
